src/main/java/com/machine/learning/classifier/Classifier.java

```diff
@@ -0,0 +1,21 @@
+package com.machine.learning.classifier;
+
+import java.util.List;
+
+public interface Classifier {
+    /**
+     * Trains the classifier on the given data
+     *
+     * @param dataPoints data to train with
+     * @param classLabel type of the class for the data
+     */
+    void train(List<List<Integer>> dataPoints, ClassLabel classLabel);
```

Comment:
I don't really think all of our input data appears as integers; it certainly seems strange to make it be that way. Maybe it could be useful for indexing, but that's it. It seems a bit annoying to convert things like "y" and "n" or "republican" and "democrat" to integers, because then any debugging output is harder to understand.

Secondly, if you're separating out the points to be classified into two arguments, don't you need a _list_ of class labels to correspond with each data point?